image_tools/src/showimage.cpp

```diff
@@ -122,31 +122,42 @@ int main(int argc, char * argv[])
       // KEEP_ALL saves all messages until they are taken.
       // KEEP_LAST enforces a limit on the number of messages that are saved,
       // specified by the "depth" parameter.
-      history_policy,
+      history_policy_,
       // Depth represents how many messages to store in history when the
       // history policy is KEEP_LAST.
-      depth
+      depth_
   ));
-
   // The reliability policy can be reliable, meaning that the underlying transport layer will try
   // ensure that every message gets received in order, or best effort, meaning that the transport
   // makes no guarantees about the order or reliability of delivery.
-  qos.reliability(reliability_policy);
-
-  auto callback = [show_camera, &node](const sensor_msgs::msg::Image::SharedPtr msg)
+  qos.reliability(reliability_policy_);
+  auto callback = [this](const sensor_msgs::msg::Image::SharedPtr msg)
     {
-      show_image(msg, show_camera, node->get_logger());
+      show_image(msg, show_camera_, this->get_logger());
     };
 
-  std::cerr << "Subscribing to topic '" << topic << "'" << std::endl;
-  RCLCPP_INFO(node->get_logger(), "Subscribing to topic '%s'", topic.c_str());
-  // Initialize a subscriber that will receive the ROS Image message to be displayed.
-  auto sub = node->create_subscription<sensor_msgs::msg::Image>(
-    topic, qos, callback);
-
-  rclcpp::spin(node);
+  RCLCPP_INFO(this->get_logger(), "Subscribing to topic '%s'", topic_.c_str());
+  sub_ = create_subscription<sensor_msgs::msg::Image>(topic_, qos, callback);
+}
 
-  rclcpp::shutdown();
 
-  return 0;
+/// Read in and parse command line arguments.
+/**
+ * \param[in] argc
+ * \param[in] argv
+ * \return A bool whether command line options were valid or not
+ */
+bool ShowImage::setup(int argc, const char ** argv)
+{
+  return parse_command_options(
+    argc, argv, &depth_, &reliability_policy_, &history_policy_, &show_camera_, nullptr, nullptr,
+    nullptr, nullptr, &topic_);
 }
+
+
```

Comment:
nit: remove extra vertical whitespace. Only one empty line is necessary (same below).